tests/venv.py

```diff
@@ -0,0 +1,33 @@
+"""
+Utilities for managing virtualenvs in tests.
+"""
+
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+
+def _use_virtualenv(venv_path, shell_script):
+    """Run `shell_script` in the already-made virtualenv at `venv_path`."""
+    subprocess.check_call(f'. {venv_path}/bin/activate; {shell_script}', env={}, shell=True)
+
+
+def run_in_virtualenv(shell_script):
+    """
+    Create a virtualenv and run the provided shell script in it.
+
+    The virtualenv is created in the current directory and deleted after
+    the script runs.
+    """
+    with TemporaryDirectory() as parent_dir:
+        pip_txt = shutil.move("requirements/pip.txt", parent_dir)
```

Comment:
Why do we need to copy/move this file rather than just pointing to the existing path? (It also seems like this can only be used once, if it's moving rather than copying.)

Reply:
Oh, yeah. :-) But also I think it's wrong either way. I'm going to approve the PR in advance but I'll also file a ticket to rework this to run tests properly, using make targets.

Reply:
I removed the copy and the extra pip.txt install.